mautic/mautic
# Special characters of the text getting corrupted while posting results to another form

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Yes
| New feature/enhancement? (use the a.x branch)      | No
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | - 
| Related developer documentation PR URL | - 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Special characters of the text getting corrupted while posting results to another form:
- If we have text field with string **`A & B`**. Upon submit the form, the destination form having the same string as **`A &#38; B`** in the linked form.
- The special character **`<`** is translated to its ASCII value of **`&#60;`** at the destination form.
- Similarly the special character **`'`** is translated to its ASCII value of **`&#39;`** at the destination form

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a form with "Post result to another form" action. (You can use webhook.site url for inspect action url)
2. Submit form's Text field with special character (Ex. **`Test's & Data`**)
3. Observe posted data to destination for is encoded/corrupted (Ex. **`Test&#39;s &#38; Data`**)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Follow the Steps to reproduce the bug.
3. Observe the posted data to destination form should be same as we submitted it from form. (Ex. **`Test's & Data`**)




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10872"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
